test/integration/policy_info_metric_test.go

```diff
@@ -0,0 +1,149 @@
+	noncompliantPolicyYaml    = "../resources/policy_info_metric/noncompliant.yaml"
+	noncompliantPolicyName    = "policy-metric-noncompliant"
+)
+
+var routeURL string
+
+var _ = Describe("Test policy_governance_info metric", func() {
+	It("Sets up the metrics service endpoint for tests", func() {
+		By("Ensuring the metrics service exists")
+		svcList, err := clientHub.CoreV1().Services(ocmNS).List(context.TODO(), metav1.ListOptions{LabelSelector: propagatorMetricsSelector})
+		Expect(err).To(BeNil())
+		Expect(len(svcList.Items)).To(Equal(1))
+		metricsSvc := svcList.Items[0]
+
+		By("Checking for an existing metrics route")
+		var routeList *unstructured.UnstructuredList
+		Eventually(func() interface{} {
+			var err error
+			routeList, err = clientHubDynamic.Resource(common.GvrRoute).Namespace(ocmNS).List(context.TODO(), metav1.ListOptions{LabelSelector: propagatorMetricsSelector})
+			if err != nil {
+				return err
+			}
+			return len(routeList.Items)
+		}, defaultTimeoutSeconds, 1).Should(Or(Equal(0), Equal(1)))
+
+		if len(routeList.Items) == 0 {
+			By("Exposing the metrics service as a route")
+			_, err = oc("expose", "service", metricsSvc.Name, "-n", ocmNS, `--overrides={"spec":{"tls":{"termination":"reencrypt"}}}`)
+			Expect(err).To(BeNil())
+
+			Eventually(func() interface{} {
+				var err error
+				routeList, err = clientHubDynamic.Resource(common.GvrRoute).Namespace(ocmNS).List(context.TODO(), metav1.ListOptions{LabelSelector: propagatorMetricsSelector})
+				if err != nil {
+					return err
+				}
+				return len(routeList.Items)
+			}, defaultTimeoutSeconds, 1).Should(Equal(1))
+		}
+
+		routeURL = routeList.Items[0].Object["spec"].(map[string]interface{})["host"].(string)
+		By("Got the metrics route url: " + routeURL)
+	})
+	It("Checks that the endpoint does not expose metrics without auth", func() {
+		Eventually(func() interface{} {
+			resp, err := curlMetricsRoute()
+			if err != nil {
+				return err
+			}
+			return resp
+		}, defaultTimeoutSeconds, 1).Should(ContainSubstring("Unauthorized"))
+	})
+	It("Checks that endpoint has a HELP comment for the metric", func() {
+		token, err := oc("whoami", "-t")
+		Expect(err).To(BeNil())
+		Eventually(func() interface{} {
+			resp, err := curlMetricsRoute("--header", "Authorization: Bearer "+token)
+			if err != nil {
+				return err
+			}
+			return resp
+		}, defaultTimeoutSeconds, 1).Should(ContainSubstring("HELP " + metricName))
+	})
+	It("Checks that a compliant policy reports a metric of 0", func() {
+		By("Creating a compliant policy")
+		oc("apply", "-f", compliantPolicyYaml, "-n", userNamespace, "--kubeconfig="+kubeconfigHub)
+		Eventually(
+			getComplianceState(compliantPolicyName),
+			defaultTimeoutSeconds,
+			1,
+		).Should(Equal(policiesv1.Compliant))
+
+		By("Checking the policy metric")
+		token, err := oc("whoami", "-t")
+		Expect(err).To(BeNil())
+		regex := `(?m)` + metricName + `{.*policy="` + compliantPolicyName + `.*} 0$`
```

Comment:
Could you please add a comment above this of the string you are capturing? This make this easier to read.

Reply:
```suggestion
		regex := `(?m)` + metricName + `{.*policy="` + compliantPolicyName + `".*} 0$`
```